exchange/exchange.go

```diff
@@ -121,9 +126,27 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 		e.me.RecordImps(impLabels)
 	}
 
+	// Make our best guess if GDPR applies
+	usersyncIfAmbiguous := e.UsersyncIfAmbiguous
+	var geo *openrtb.Geo = nil
+	if bidRequest.User != nil && bidRequest.User.Geo != nil {
+		geo = bidRequest.User.Geo
+	} else if bidRequest.Device != nil && bidRequest.Device.Geo != nil {
+		geo = bidRequest.Device.Geo
+	}
+	if geo != nil {
+		// If we have a country set, and it is on the list, we assume GDPR applies if not set on the request.
+		// Otherwise we assume it does not apply as long as it appears "valid" (is 3 characters long).
+		if _, found := e.eeaCountries[strings.ToUpper(geo.Country)]; found {
+			usersyncIfAmbiguous = false
+		} else if len(geo.Country) == 3 {
+			// The country field is formatted properly as a three character country code
+			usersyncIfAmbiguous = true
+		}
+	}
```

Comment:
Just to make it extra readable maybe we could refactor this entire block to live its own function
```
129     // Make our best guess if GDPR applies
    +   usersyncIfAmbiguous := processUserSyncEnforcement(bidRequest, e)
130 -   usersyncIfAmbiguous := e.UsersyncIfAmbiguous
131 -   var geo *openrtb.Geo = nil
132 -   if bidRequest.User != nil && bidRequest.User.Geo != nil {
133 -       geo = bidRequest.User.Geo
134 -   } else if bidRequest.Device != nil && bidRequest.Device.Geo != nil {
135 -       geo = bidRequest.Device.Geo
136 -   }
137 -   if geo != nil {
138 -       // If we have a country set, and it is on the list, we assume GDPR applies if not set on the request.
139 -       // Otherwise we assume it does not apply as long as it appears "valid" (is 3 characters long).
140 -       if _, found := e.eeaCountries[strings.ToUpper(geo.Country)]; found {
141 -           usersyncIfAmbiguous = false
142 -       } else if len(geo.Country) == 3 {
143 -           // The country field is formatted properly as a three character country code
144 -           usersyncIfAmbiguous = true
145 -       }
146 -   }
exchange/exchange.go
```